cmd/status.go

```diff
@@ -551,50 +562,94 @@ func displayFailedEntries(client *server.Client, given *set.Set) {
 
 // displayAllEntries prints out details about all entries in the given
 // set.
-func displayAllEntries(client *server.Client, given *set.Set) {
+func displayAllEntries(client *server.Client, given *set.Set, showRemotePaths bool, transformer put.PathTransformer) {
 	all, err := client.GetFiles(given.ID())
 	if err != nil {
 		die(err.Error())
 	}
 
-	displayEntries(all)
+	displayEntries(all, showRemotePaths, transformer)
 }
 
 // displayEntries prints info about the given file entries to STDOUT.
-func displayEntries(entries []*set.Entry) {
+func displayEntries(entries []*set.Entry, showRemotePaths bool, transformer put.PathTransformer) {
 	if len(entries) == 0 {
 		return
 	}
 
-	printEntriesHeader()
+	displayHeader(showRemotePaths)
+
+	warnedAboutTransformer := false
 
 	for _, entry := range entries {
-		var date string
+		var remotePath string
 
-		if entry.LastAttempt.IsZero() {
-			date = "-"
-		} else {
-			date = entry.LastAttempt.Format(dateShort)
+		if showRemotePaths {
+			remotePath, warnedAboutTransformer = getRemotePath(entry.Path, transformer, warnedAboutTransformer)
 		}
 
-		cols := []string{
-			entry.Path,
-			entry.Status.String(),
-			humanize.IBytes(entry.Size),
-			fmt.Sprintf("%d", entry.Attempts),
-			date,
-			entry.LastError,
-		}
+		displayEntry(entry, remotePath)
+	}
+}
+
+// displayHeader adds a column for remote path if showRemotePath is true and
+// prints the header.
+func displayHeader(showRemotePath bool) {
+	cols := []string{"Path", "Status", "Size", "Attempts", "Date", "Error"}
 
-		cliPrintRaw(strings.Join(cols, "\t"))
-		cliPrintRaw("\n")
+	if showRemotePath {
+		cols = append(cols, "Remote Path")
 	}
+
+	printEntriesHeader(cols)
 }
 
-// printEntriesHeader prints a header for a subsequent 6 column output of entry
-// details.
-func printEntriesHeader() {
+// printEntriesHeader prints a header including the given columns relating to
+// the output of entry details.
+func printEntriesHeader(cols []string) {
 	cliPrint("\n")
-	cliPrint(strings.Join([]string{"Path", "Status", "Size", "Attempts", "Date", "Error"}, "\t"))
+	cliPrint(strings.Join(cols, "\t"))
 	cliPrint("\n")
 }
+
+// getRemotePath returns the remote path for a given path.
+func getRemotePath(path string, transformer put.PathTransformer, warnedAboutTransformer bool) (string, bool) {
+	remotePath, err := transformer(path)
+	if err != nil {
+		if !warnedAboutTransformer {
+			warn("your transformer didn't work: %s", err)
```

Comment:
This should probably just die, and then you don't have to return and supply the warnedAbout bool.